src/test/java/at/ac/tuwien/kr/alpha/common/fixedinterpretations/FixedInterpretationLiteralsTest.java

```diff
@@ -0,0 +1,199 @@
+
+import at.ac.tuwien.kr.alpha.api.Alpha;
+import at.ac.tuwien.kr.alpha.api.externals.Externals;
+import at.ac.tuwien.kr.alpha.api.externals.stdlib.AspStandardLibrary;
+import at.ac.tuwien.kr.alpha.common.AnswerSet;
+import at.ac.tuwien.kr.alpha.common.Predicate;
+import at.ac.tuwien.kr.alpha.common.terms.ConstantTerm;
+
+public class FixedInterpretationLiteralsTest {
+
+	@at.ac.tuwien.kr.alpha.api.externals.Predicate
+	public static boolean isOne(int value) {
+		return value == 1;
+	}
+
+	/**
+	 * Dummy method to test external atoms with more than one output term list.
+	 * Dummy input param which is not used exists solely to avoid producing an edge
+	 * case
+	 * where there is output, but no input
+	 */
+	@at.ac.tuwien.kr.alpha.api.externals.Predicate
+	public static Set<List<ConstantTerm<String>>> connection(String dummy) {
+		Set<List<ConstantTerm<String>>> retVal = new HashSet<>();
+		List<ConstantTerm<String>> l1 = new ArrayList<>();
+		List<ConstantTerm<String>> l2 = new ArrayList<>();
+		List<ConstantTerm<String>> l3 = new ArrayList<>();
+		l1.add(ConstantTerm.getInstance("Klagenfurt"));
+		l1.add(ConstantTerm.getInstance("Villach"));
+		l2.add(ConstantTerm.getInstance("Klagenfurt"));
+		l2.add(ConstantTerm.getInstance("Graz"));
+		l3.add(ConstantTerm.getInstance("Villach"));
+		l3.add(ConstantTerm.getInstance("Salzburg"));
+		retVal.add(l1);
+		retVal.add(l2);
+		retVal.add(l3);
+		return retVal;
+	}
+
+	private static final String TEST_PROG = "positive_numeric_comparison :- 2 < 3."
+			+ "negative_numeric_comparison :- not 3 < 2."
+			+ "positive_unary_external :- &isOne[1]."
+			+ "negative_unary_external :- not &isOne[2]."
+			+ "positive_external_with_output :- "
+			+ "&stdlib_datetime_parse[\"01.01.2121 21:11:12\", \"dd.MM.yyyy HH:mm:ss\"](Y, MO, D, H, MI, S), Y = 2121, MO = 1, D = 1, H = 21, MI = 11, S = 12."
+			+ "positive_external_with_output_dontfire :- "
+			+ "&stdlib_datetime_parse[\"01.01.2121 21:11:12\", \"dd.MM.yyyy HH:mm:ss\"](Y, MO, D, H, MI, S), Y = 2345, MO = 1, D = 1, H = 21, MI = 11, S = 12."
+			+ "negative_external_with_output :- Y = 2121, MO = 1, D = 1, H = 21, MI = 11, S = 12, "
+			+ "not &stdlib_datetime_parse[\"22.02.2222 21:11:12\", \"dd.MM.yyyy HH:mm:ss\"](Y, MO, D, H, MI, S)."
+			+ "negative_external_with_output_dontfire :- Y = 2222, MO = 2, D = 22, H = 21, MI = 11, S = 12,"
+			+ "not &stdlib_datetime_parse[\"22.02.2222 21:11:12\", \"dd.MM.yyyy HH:mm:ss\"](Y, MO, D, H, MI, S)."
+			+ "negative_external_multioutput :- START = \"Graz\", END = \"Salzburg\", not &connection[train_network](START, END)."
+			+ "negative_external_multioutput_dontfire :- START = \"Klagenfurt\", END = \"Villach\", not &connection[train_network](START, END)."
+			+ "positive_external_multioutput :- START = \"Klagenfurt\", END = \"Villach\", &connection[train_network](START, END)."
+			+ "positive_external_multioutput_dontfire :- START = \"Graz\", END = \"Salzburg\", &connection[train_network](START, END).";
```

Comment:
This is very good and tests many combinations!
I think, an important case is missing, however: one where the external literal is guaranteed to generate variable bindings (here, the evaluation order could always bind the output variables to some constant, before evaluating the external literal. So, I think a rule where some variable outputs from the external literal are fed into the head of the rule should be added (and checked). For example:
`"external_literal_output(START, END) :-  &connection[train_network](START, END)."`

Reply:
Thanks for catching that! Added a test case to cover that..